dev/breeze/src/airflow_breeze/cache.py

```diff
@@ -0,0 +1,77 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+from typing import Any, List, Tuple
+
+from airflow_breeze import global_constants
+from airflow_breeze.breeze import get_airflow_sources_root
+
+# This file is to read the cached data in .build folder
+BUILD_CACHE_DIR = os.path.join(get_airflow_sources_root, '.build')
+
+
+def read_from_cache_file(param_name: str):
+    parameter_value = ""
+    parameter_value_file = os.path.join(BUILD_CACHE_DIR, f".{param_name}")
+    with open(parameter_value_file) as value:
+        parameter_value = value.read()
+    return parameter_value
+
+
+def write_to_cache_file(param_name: str, param_value: str, check_allowed_parameter: bool = True):
+    allowed = False
+    if check_allowed_parameter:
+        allowed, allowed_values = check_if_parameter_allowed(param_name, param_value)
+    if (check_allowed_parameter is False) or ((check_allowed_parameter is True) and allowed):
+        # Then write to cache file after (verifying and allowed) or (without any verification)
+        parameter_value_file = os.path.join(BUILD_CACHE_DIR, f".{param_name}")
+        with open(parameter_value_file, 'w') as value:
+            value.write(param_value)
+    else:
+        # Supported param value is not provided.
+        print('Write to cache dir failed.')
+
+
+def check_cache_and_write_if_not_cached(param_name: str, default_param_value: str):
+    is_cached = False
+    allowed = False
+    cached_value = read_from_cache_file(param_name)
+    if cached_value == "":
+        write_to_cache_file(param_name, default_param_value)
+    else:
+        allowed, _ = check_if_parameter_allowed(param_name, cached_value)
+        if allowed:
+            is_cached = True
+        else:
+            write_to_cache_file(param_name, default_param_value)
+    return is_cached, cached_value
+
+
+# Check if the passed parameter is allowed value in the list of supported params
+def check_if_parameter_allowed(param_name: str, param_value: str) -> Tuple[bool, List[Any]]:
+    allowed = False
+    allowed_values: List[Any] = []
+    allowed_values = getattr(global_constants, f'ALLOWED_{param_name.upper()}')
+    if param_value in allowed_values:
+        # The value passed is allowed
+        allowed = True
+    else:
+        print(f'You have sent the {param_value} for {param_name}')
```

Comment:
I think this information could be doen in the caller (see the other comment). We might want to use that function for various checks and sometimes we do no want to print that information - simply returning the "allowed" + what allowed_values are should do the job and we should take care about messaging when the function returns.